airflow/serialization/serialized_dag.py

```diff
@@ -101,7 +102,7 @@ def deserialize_dag(cls, encoded_dag: dict) -> "SerializedDAG":
         setattr(dag, 'full_filepath', dag.fileloc)
         for task in dag.task_dict.values():
             task.dag = dag
-            serializable_task: SerializedBaseOperator = task
+            serializable_task: SerializedBaseOperator = cast(SerializedBaseOperator, task)
```

Comment:
Is this required?

Reply:
Yes. MyPy complains if it's not there. Maybe it's worth looking whether there is a good reason for that @kaxil ?

Reply:
I will shortly

Reply:
Cool, other than that I am fine with this PR :)

Thanks @potiuk

Reply:
I think it's worth looking at it @kaxil -> there are still some parts of the code that needs some more typing to catch potential errors (DagBag especially) but this one looks a bit fishy to me.

Reply:
I think the root cause is that .subdag field is only set in SerializedBaseOperator and that's why the type is forced here.